py/builtinimport.c

```diff
@@ -237,6 +242,15 @@ mp_obj_t mp_builtin___import__(mp_uint_t n_args, const mp_obj_t *args) {
     }
     DEBUG_printf("Module not yet loaded\n");
 
+    #if MICROPY_MODULE_FROZEN
+    mp_lexer_t *lex = mp_find_frozen_module(mod_str, mod_len);
+    if (lex != NULL) {
+        module_obj = mp_obj_new_module(MP_OBJ_QSTR_VALUE(module_name));
```

Comment:
Probably not.  On line 227 the qstr is extracted from module_name using mp_obj_str_get_qstr, so should probably cache that return value and use it here.